guardian/manage-frontend
# Update MMA to handle `members-data-api` response with user details section added

## What does this change?
members data api response body has been refactored so user details section added. A new object has been created called MembersDataApiResponse , this is made up of user (firstName, lastName and email) and products (MembersDataApiItem[]).

This change needs to be released carefully as it relies on the server side change to link up which is why the calls to the API /api/me/mma have been refactored to accept the new object, MembersDataApiResponse, and the current working. 
MembersDataApiResponse has been added to all of the tests where there is a call to /api/me/mma. 

## How to test
On CODE and PROD manually check the screens on the manage-frontend repo and make sure that all the screens and functionality which call out to  /api/me/mma are working.

## How can we measure success?
Success will look like the journeys on manage frontend are not impacted by this change and future tickets in the backlog which require this change are unblocked.

## Have we considered potential risks?
As there is linking up required with a server side change there is a risk that it doesn't work. This can be mitigated with making sure that the first change (i.e. the PR) works on CODE and PROD. Once we are satisfied we can look to release the server side change and make sure that there are no errors or impact on calls to /api/me/mma. Next step is to remove the old object and follow the same steps to check it is working as expected.

## Images



## Accessibility
